src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +61,723 @@ const ActorInit En_Fz_InitVars = {
+        primAlpha = 150;
+        func_8002F974(&this->actor, NA_SE_EN_FREEZAD_BREATH - SFX_FLAG);
+        if ((this->counter & 0x3F) >= 48) {
+            primAlpha = 630 - ((this->counter & 0x3F) * 10);
+        }
+
+        accel.x = accel.z = 0.0f;
+        accel.y = 0.6f;
+        pos.x = this->actor.posRot.pos.x;
+        pos.y = this->actor.posRot.pos.y + 20.0f;
+        pos.z = this->actor.posRot.pos.z;
+        Matrix_RotateY((this->actor.shape.rot.y / (f32)0x8000) * M_PI, MTXMODE_NEW);
+        vec1.x = 0.0f;
+        vec1.y = -2.0f;
+        vec1.z = 20.0f;
+        Matrix_MultVec3f(&vec1, &velocity);
+        if ((this->counter % 8) == 0) {
+            isTimerMod8 = true;
+        }
+
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, isTimerMod8);
+        pos.x += (velocity.x * 0.5f);
+        pos.y += (velocity.y * 0.5f);
+        pos.z += (velocity.z * 0.5f);
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, false);
+    }
+}
+
+void EnFz_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnFz* this = THIS;
+    s32 pad;
+    static EnFzSpawnIceSmokeFunc SpawnIceSmoke[] = {
+        EnFz_SpawnIceSmokeHiddenState,  // Hiding
+        EnFz_SpawnIceSmokeGrowingState, // Fully grown
+        EnFz_SpawnIceSmokeActiveState,  // Growing or Shrinking to/from hiding
+        EnFz_SpawnIceSmokeActiveState,  // melting from fire attack
+    };
+
+    this->counter++;
+    if (this->unusedTimer1) {
+        this->unusedTimer1--;
+    }
+    if (this->timer) {
+        this->timer--;
+    }
+    if (this->unusedTimer2) {
+        this->unusedTimer2--;
+    }
+
+    Actor_SetHeight(&this->actor, 50.0f);
+    EnFz_ApplyDamage(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+    if (this->isDespawning == false) {
+        Collider_CylinderUpdate(&this->actor, &this->collider1);
+        Collider_CylinderUpdate(&this->actor, &this->collider2);
+        if (this->isFreezing) {
+            if (this->actor.dmgEffectTimer == 0) {
+                CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider1);
+                CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider2);
+            }
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider1);
+        }
+    }
+
+    Math_StepToF(&this->actor.speedXZ, this->speedXZ, 0.2f);
+    Actor_MoveForward(&this->actor);
+    if (this->isAlwaysTrue) {
+        func_8002E4B4(globalCtx, &this->actor, 20.0f, 20.0f, 20.0f, 5);
+    }
+
+    SpawnIceSmoke[this->state](this);
+    EnFz_MoveIceSmoke(this, globalCtx);
+}
+
+void EnFz_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Gfx* displayLists[] = {
+        0x06001130, // Body fully intact           (5 or 6 health)
+        0x060021A0, // Top right horn chipped off  (from Freezards perspective)   (3 or 4 health)
+        0x06002CA0
+    }; // Entire head chipped off     (1 or 2 health)
```

Comment:
```suggestion
        0x06002CA0, // Entire head chipped off     (1 or 2 health)
    }; 
```
formats better like this, right?